### PR TITLE
BUG: import of maybe_convert_indices in pandas.core.index.py, #10610

### LINKED ISSUE #10610: Naming inconsistency in import of maybe_convert_indices in pandas.core.index.py
In lines 930-931 of pandas.core.index.py, the following code states
```python
from pandas.core.indexing import _maybe_convert_indices
return _maybe_convert_indices(indexer, len(self))
```
In pandas.core.indexing.py, the function is called `maybe_convert_indices`. In the 3 other places where the function is called, it is called using the correct name. The leading underscore should be removed from the function name in index.py.

### PR DESCRIPTION
**Pull request by djgagne**
closes #10610 

I fixed the import statement and added a test to check for proper behavior when accessing a mixed-integer index with a list of values.

### CODE REVIEW

File: pandas/core/index.py [hunk 1]
```diff
@@ -971,11 +971,13 @@ def _convert_list_indexer(self, keyarr, kind=None):
 
             if self.inferred_type == 'mixed-integer':
                 indexer = self.get_indexer(keyarr)
+                if not np.all(np.in1d(indexer, self.values)):
```
Comment (jreback) [inline pandas/core/index.py]: why is this line necessary?

Comment (djgagne) [inline pandas/core/index.py]: Without the line, when passed a list containing index values, the function would output the data associated with the nearest integer index value. For other cases where an invalid index is passed to a DataFrame, an IndexError would be raised, so I made sure to explicitly check for that condition.

Comment (jreback) [inline pandas/core/index.py]: your test doesn't hit that line though it goes thru
can u make a test that is explicitly caught there?

**Comment by djgagne** [comment]
I removed the index checking if-statement, and added a line that would change the -1 indices to equal the length of the Index being evaluated, so that the IndexError would be raised in maybe_convert_indices.

File: pandas/core/index.py [hunk 2]
```diff
@@ -973,9 +973,12 @@ def _convert_list_indexer(self, keyarr, kind=None):
...
                     return indexer
-
-                from pandas.core.indexing import _maybe_convert_indices
-                return _maybe_convert_indices(indexer, len(self))
+                # missing values are flagged as -1 by get_indexer and negative indices are already
+                # converted to positive indices in the above if-statement, so the negative flags are changed to
+                # values outside the range of indices so as to trigger an IndexError in maybe_convert_indices
+                indexer[indexer < 0] = len(self)
+                from pandas.core.indexing import maybe_convert_indices
+                return maybe_convert_indices(indexer, len(self))
```
Comment (jreback) [inline pandas/core/index.py]: this is already done inside ``maybe_convert_indices`` why do you think its needed here?

Comment (djgagne) [inline pandas/core/index.py]: It is necessary because negative indices passed into maybe_convert_indices are summed by the length of the array before the bounds checking is done. Because get_indexer assigns labels that are not found in the index with a value of -1, the last row or column is assigned to those positions instead of either NaNs or an exception being raised. With the line I added, the bounds checking in maybe_convert_indices will now throw an IndexError for out of bounds indices.

**Comment by jreback** [comment]
ok, this is fine.

pls squash to a single commit. ping when green.

File: pandas/tests/test_index.py [hunk 3]
```diff
@@ -1729,6 +1729,11 @@ def test_equals_op_multiindex(self):
             df.index == index_a
         tm.assert_numpy_array_equal(index_a == mi3, np.array([False, False, False]))
 
+    def test_multitype_list_index_access(self):
+        df = pd.DataFrame(np.random.random((10, 5)), columns=["a"] + [20, 21, 22, 23])
```
Comment (jreback) [inline pandas/tests/test_index.py]: add the issue number as a comment here

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/12df3ea39b28f752a0dcb43b87c4121387114cdf

thanks!
